src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -114,20 +113,25 @@
     @VisibleForTesting
     static Iterable<AlignmentRegion> breakGappedAlignment(final AlignmentRegion oneRegion, final int sensitivity) {
 
-        final List<CigarElement> cigarElements = oneRegion.forwardStrandCigar.getCigarElements();
+        final List<CigarElement> cigarElements = new ArrayList<>();
+        final boolean cigarHasInsertionAsTrailingOperator = parseCigarAndConvertTrailingInsertionToSoftClip(oneRegion.cigarAlong5to3DirectionOfContig, cigarElements);
         if (cigarElements.size() == 1) return new ArrayList<>( Collections.singletonList(oneRegion) );
 
         final List<AlignmentRegion> result = new ArrayList<>(10); // blunt guess
         final int originalMapQ = oneRegion.mapQual;
 
         final List<CigarElement> cigarMemoryList = new ArrayList<>();
-        final int clippedNBases = SVVariantCallerUtils.getNumClippedBases(true, oneRegion.forwardStrandCigar);
+        int clippedNBasesFromStart = SVVariantCallerUtils.getNumClippedBases(true, oneRegion.cigarAlong5to3DirectionOfContig);
+        if (cigarHasInsertionAsTrailingOperator && cigarElements.get(0).getOperator()==CigarOperator.S) {
+            clippedNBasesFromStart = cigarElements.get(0).getLength();
+        }
         final int hardClippingAtBeginning = cigarElements.get(0).getOperator()==CigarOperator.H ? cigarElements.get(0).getLength() : 0;
         final int hardClippingAtEnd = (cigarElements.get(cigarElements.size()-1).getOperator()== CigarOperator.H)? cigarElements.get(cigarElements.size()-1).getLength() : 0;
         final CigarElement hardClippingAtBeginningMaybeNull = hardClippingAtBeginning==0 ? null : new CigarElement(hardClippingAtBeginning, CigarOperator.H);
-        int     refIntervalStart    = oneRegion.referenceInterval.getStart(),
-                contigIntervalStart = 1 + clippedNBases;
+        int contigIntervalStart = 1 + clippedNBasesFromStart;
         int gapCount = 0;
+        // we are walking along the contig following the cigar, which indicates that we might be walking backwards on the reference if oneRegion.forwardStrand==false
+        int refBoundary1stInTheDirectionOfContig = oneRegion.forwardStrand ? oneRegion.referenceInterval.getStart() : oneRegion.referenceInterval.getEnd();
```

Comment:
If we reversed the cigar here first, would we be able to get rid of all of the if statements in the body of this loop? It seems like it might make the code a lot cleaner. But I guess you'd still need to keep track of it for the on-contig coordinates. Just something to think about.